programs/locker/src/util/token2022.rs

```diff
@@ -0,0 +1,250 @@
+use anchor_spl::token_interface::{Mint, TokenAccount, TokenInterface};
+
+use crate::{LockerError, VestingEscrow};
+
+const ONE_IN_BASIS_POINTS: u128 = MAX_FEE_BASIS_POINTS as u128;
+
+#[derive(Clone, Copy)]
+pub struct MemoTransferContext<'a, 'info> {
+    pub memo_program: &'a Program<'info, Memo>,
+    pub memo: &'static [u8],
+}
+
+pub fn transfer_to_escrow_v2<'a, 'c: 'info, 'info>(
+    sender: &'a Signer<'info>,
+    token_mint: &InterfaceAccount<'info, Mint>,
+    sender_token: &InterfaceAccount<'info, TokenAccount>,
+    escrow_token: &InterfaceAccount<'info, TokenAccount>,
+    token_program: &Interface<'info, TokenInterface>,
+    amount: u64,
+) -> Result<()> {
+    let instruction = spl_token_2022::instruction::transfer_checked(
+        token_program.key,
+        &sender_token.key(),
+        &token_mint.key(),   // mint
+        &escrow_token.key(), // to
+        sender.key,          // authority
+        &[],
+        // The transfer amount should include fee
+        amount,
+        token_mint.decimals,
+    )?;
+
+    let account_infos = vec![
+        token_program.to_account_info(),
+        sender_token.to_account_info(),
+        token_mint.to_account_info(),
+        escrow_token.to_account_info(),
+        sender.to_account_info(),
+    ];
+
+    solana_program::program::invoke(&instruction, &account_infos)?;
+
+    Ok(())
+}
+
+pub fn transfer_to_user_v2<'c: 'info, 'info>(
+    escrow: &AccountLoader<'info, VestingEscrow>,
+    token_mint: &InterfaceAccount<'info, Mint>,
+    escrow_token: &InterfaceAccount<'info, TokenAccount>,
+    recipient_account: &InterfaceAccount<'info, TokenAccount>,
+    token_program: &Interface<'info, TokenInterface>,
+    memo_transfer_context: Option<MemoTransferContext<'_, 'info>>,
+    amount: u64,
+) -> Result<()> {
+    let escrow_state = escrow.load()?;
+    let escrow_seeds = escrow_seeds!(escrow_state);
+
+    if let Some(memo_ctx) = memo_transfer_context {
+        if is_transfer_memo_required(&recipient_account)? {
+            memo::build_memo(
+                CpiContext::new(memo_ctx.memo_program.to_account_info(), BuildMemo {}),
+                memo_ctx.memo,
+            )?;
+        }
+    }
+
+    let instruction = spl_token_2022::instruction::transfer_checked(
+        token_program.key,
+        &escrow_token.key(),
+        &token_mint.key(),        // mint
+        &recipient_account.key(), // to
+        &escrow.key(),            // authority
+        &[],
+        amount,
+        token_mint.decimals,
+    )?;
+
+    let account_infos = vec![
+        escrow_token.to_account_info(),
+        token_mint.to_account_info(),
+        recipient_account.to_account_info(),
+        escrow.to_account_info(),
+    ];
+
+    solana_program::program::invoke_signed(&instruction, &account_infos, &[&escrow_seeds[..]])?;
+
+    Ok(())
+}
+
+pub fn validate_mint(token_mint: &InterfaceAccount<Mint>) -> Result<bool> {
```

Comment:
all function with prefix validate should return Result<()>

Reply:
updated